core/http-auth-aws-event-stream/src/main/java/software/amazon/awssdk/http/auth/aws/eventstream/signer/DefaultAwsV4EventStreamHttpSigner.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.eventstream.signer;
+
+import static software.amazon.awssdk.http.auth.aws.util.SignerConstant.X_AMZ_CONTENT_SHA256;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.CompletableFuture;
+import org.reactivestreams.Publisher;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.http.auth.aws.checksum.ContentChecksum;
+import software.amazon.awssdk.http.auth.aws.checksum.SdkChecksum;
+import software.amazon.awssdk.http.auth.aws.eventstream.AwsV4EventStreamHttpSigner;
+import software.amazon.awssdk.http.auth.aws.eventstream.internal.SigV4DataFramePublisher;
+import software.amazon.awssdk.http.auth.aws.signer.AwsV4CanonicalRequest;
+import software.amazon.awssdk.http.auth.aws.signer.AwsV4Properties;
+import software.amazon.awssdk.http.auth.aws.signer.BaseAwsV4HttpSigner;
+import software.amazon.awssdk.http.auth.aws.signer.SigV4Context;
+import software.amazon.awssdk.http.auth.spi.AsyncSignRequest;
+import software.amazon.awssdk.http.auth.spi.SignRequest;
+import software.amazon.awssdk.http.auth.spi.SyncSignRequest;
+import software.amazon.awssdk.http.auth.spi.SyncSignedRequest;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+
+/**
+ * A default implementation of {@link AwsV4EventStreamHttpSigner}.
+ * TODO: Rename this correctly once the interface is gone and checkstyle can pass
+ */
+@SdkProtectedApi
+public final class DefaultAwsV4EventStreamHttpSigner implements BaseAwsV4HttpSigner<AwsV4Properties> {
```

Comment:
Why does the implementation need to be protected?

Reply:
Because it's used outside of the module, in the DefaultAwsV4HttpSigner, where the delegation happens.